fenix/app/src/main/java/org/mozilla/fenix/messaging/state/MessagingMiddleware.kt

```diff
@@ -37,7 +38,9 @@ class MessagingMiddleware(
             is Restore -> {
                 coroutineScope.launch {
                     val messages = controller.getMessages()
-                    context.store.dispatch(UpdateMessages(messages))
+                    MainScope().launch {
+                        context.store.dispatch(UpdateMessages(messages))
```

Comment:
Looking a the previous implementation we where just calling `context.dispatch(UpdateMessages(messages))
` instead of `context.store.dispatch`
https://github.com/mozilla-mobile/firefox-android/pull/5699/files#diff-05e0a602e09d4173a4b88f3854cedbb3d7cdf322902761aa827a337e88ccdceaL82

Reply:
No, I tried that and it consistently crashes the app on startup.

Reply:
👍🏽

Reply:
Thanks for the clarification!

Reply:
Actually, I think what I need to do is the opposite.  I think I have a much safer fix now.  I'll update.